From 47ebf589250159d6a98bd91625aefd2e1383ec26 Mon Sep 17 00:00:00 2001
From: Oliver Reiche <oliver.reiche@gmail.com>
Date: Wed, 3 May 2023 12:57:06 +0200
Subject: [PATCH] Fix missing oid type for "fake" repositories

... otherwise git_tree__parse_raw() will fail to obtain
the correct oid size, which causes the entire parse to fail.
---
 src/libgit2/repository.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libgit2/repository.c b/src/libgit2/repository.c
index 804e436abeb..c45c5855058 100644
--- a/src/libgit2/repository.c
+++ b/src/libgit2/repository.c
@@ -1152,6 +1152,8 @@ int git_repository_wrap_odb(git_repository **repo_out, git_odb *odb)
 	repo = repository_alloc();
 	GIT_ERROR_CHECK_ALLOC(repo);
 
+	repo->oid_type = GIT_OID_DEFAULT;
+
 	git_repository_set_odb(repo, odb);
 	*repo_out = repo;
 
